File: backend/src/main/java/com/carex/service/UserService.java
package com.carex.service;

import com.carex.entity.User;
import com.carex.entity.enums.Role;

import java.util.List;
import java.util.Optional;

public interface UserService {
    User createUser(String name, String email, String passwordHash, Role role, String phone);
    User getUserById(Long id);
    Optional<User> findByEmail(String email);
    List<User> getAllUsers();
    List<User> getUsersByRole(Role role);
    User updateUser(Long id, String name, String phone);
    User setActiveStatus(Long id, boolean active);
    void deleteUser(Long id);
}

File: frontend/src/services/userService.js
import api from './api';

export const userService = {
  getAllUsers: async () => {
    const response = await api.get('/users');
    return response.data;
  },

  getUserById: async (id) => {
    const response = await api.get(`/users/${id}`);
    return response.data;
  },

  updateUser: async (id, data) => {
    const response = await api.put(`/users/${id}`, data);
    return response.data;
  },

  deactivateUser: async (id) => {
    const response = await api.delete(`/users/${id}`);
    return response.data;
  },
};

export default userService;
